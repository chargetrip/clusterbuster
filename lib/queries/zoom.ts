/**
 * @description The dafault implementation of zoom to distance
 */
export const defaultZoomToDistance = (zoomLevel: number, radius: number = 15) =>
  radius / Math.pow(2, zoomLevel);
