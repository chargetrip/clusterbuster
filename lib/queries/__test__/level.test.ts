import {
  defaultGetLevelClusterQuery,
  defaultGetLevelGroupQuery,
} from '../level';
import { defaultZoomToDistance } from '../zoom';

describe('defaultGetLevelClusterQuery', () => {
  it('should work', () => {
    expect(
      defaultGetLevelClusterQuery({
        parentTable: 'base_query',
        zoomLevel: 14,
        radius: 10,
        attributes: ', id, name',
        zoomToDistance: defaultZoomToDistance,
      })
    ).toMatchSnapshot();
  });
});

describe('defaultGetLevelGroupQuery', () => {
  it('should work', () => {
    expect(
      defaultGetLevelGroupQuery({
        zoomLevel: 14,
        attributes: 'FIRST(id), FIRST(name), ',
      })
    ).toMatchSnapshot();
  });
});
