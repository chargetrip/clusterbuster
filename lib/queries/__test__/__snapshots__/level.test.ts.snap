// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultGetLevelClusterQuery should work 1`] = `
"
SELECT
  center,
  expansionZoom,
  clusterNo AS previousClusterNo,
  size,
  ST_ClusterDBSCAN(center, 0.0006103515625, 1) over () as clusters, id, name
FROM base_query
"
`;

exports[`defaultGetLevelGroupQuery should work 1`] = `
"
SELECT
  SUM(size) as size,
  clusters AS clusterNo,
  (
    CASE COUNT(previousClusterNo) 
      WHEN 1 THEN FIRST(expansionZoom) 
      ELSE 15 END
  ) AS expansionZoom, FIRST(id), FIRST(name), 
  ST_Centroid(ST_Collect(center)) as center
FROM clustered_14
GROUP BY clusters
"
`;
